cmd/kubeadm/app/phases/kubelet/flags.go

```diff
@@ -78,7 +79,12 @@ func buildKubeletArgMap(nodeRegOpts *kubeadmapi.NodeRegistrationOptions, registe
 		kubeletFlags["resolv-conf"] = "/run/systemd/resolve/resolv.conf"
 	}
 
-	// TODO: Pass through --hostname-override if a custom name is used?
+	// Make sure the node name we're passed will work with Kubelet
+	if nodeRegOpts.Name != "" && nodeRegOpts.Name != nodeutil.GetHostname("") {
```

Comment:
given that the most common thing folks do is use fully qualified domain names I'm almost wondering if we should force tolower on the names similar to your other patch.

Reply:
I just tested this, and it appears this already happens automatically:

`ubuntu@ip-172-31-84-104:~$ sudo /tmp/kubeadm init -v 1 --pod-network-cidr=192.168.0.0/16 --node-name ALLCAPS.horse`

```
ubuntu@ip-172-31-84-104:~$ kubectl get nodes
NAME            STATUS    ROLES     AGE       VERSION
allcaps.horse   Ready     master    44s       v1.11.0-beta.0
```

I don't particularly know how, but I think this is a nonissue either way.

Reply:
The `nodeutil.GetHostname()` func lowercases automatically:

```go
// GetHostname returns OS's hostname if 'hostnameOverride' is empty; otherwise, return 'hostnameOverride'.
func GetHostname(hostnameOverride string) string {
	hostname := hostnameOverride
	if hostname == "" {
		nodename, err := os.Hostname()
		if err != nil {
			glog.Fatalf("Couldn't determine hostname: %v", err)
		}
		hostname = nodename
	}
	return strings.ToLower(strings.TrimSpace(hostname))
}
```